db/database_test.go

```diff
@@ -1240,34 +1240,37 @@ func TestRecentSequenceHistory(t *testing.T) {
 	defer testBucket.Close()
 	defer tearDownTestDB(t, db)
 
+	seqTracker := uint64(0)
+
 	// Validate recent sequence is written
 	body := Body{"val": "one"}
 	revid, err := db.Put("doc1", body)
+	seqTracker++
 
+	expectedRecent := make([]uint64, 0)
 	assert.True(t, revid != "")
 	doc, err := db.GetDocument("doc1", DocUnmarshalAll)
+	expectedRecent = append(expectedRecent, seqTracker)
 	assert.True(t, err == nil)
-	assert.DeepEquals(t, doc.RecentSequences, []uint64{1})
+	assert.DeepEquals(t, doc.RecentSequences, expectedRecent)
 
-	// Add a few revisions - validate they are retained when less than max (20)
-	for i := 0; i < 3; i++ {
+	// Add kMaxRecentSequences more revisions - validate they are retained when total is less than max
+	for i := 0; i < kMaxRecentSequences; i++ {
 		revid, err = db.Put("doc1", body)
+		seqTracker++
+		expectedRecent = append(expectedRecent, seqTracker)
 	}
 
 	doc, err = db.GetDocument("doc1", DocUnmarshalAll)
 	assert.True(t, err == nil)
-	assert.DeepEquals(t, doc.RecentSequences, []uint64{1, 2, 3, 4})
+	assert.DeepEquals(t, doc.RecentSequences, expectedRecent)
 
-	// Add many sequences to validate pruning when past max (20)
-	for i := 0; i < kMaxRecentSequences; i++ {
-		revid, err = db.Put("doc1", body)
-		// Sleep needed to ensure consistent results when running single-threaded vs. multi-threaded test:
-		// without it we can't predict the relative update times of nextSequence and RecentSequences
-		time.Sleep(50 * time.Millisecond)
-	}
+	// Wait for db.sequences.nextSequence to catch up (used by pruning calculation)
```

Comment:
I don't fully understand this comment .. can you add a bit more explaining why it needs to catch up?

Reply:
Would this also work?  `db.changeCache.waitForSequence(seqTracker)`?

Reply:
> // Wait for db.sequences.nextSequence to catch up (used by pruning calculation)

Oh, so basically the change tracker needs to catch up on the async dcp feed?

Reply:
Good point - should use seqTracker here.